libraries/AP_HAL_Linux/Scheduler.cpp

```diff
@@ -253,6 +253,11 @@ void Scheduler::resume_timer_procs()
     _timer_semaphore.give();
 }
 
+void Scheduler::_poller_task()
+{
+    _poller.poll(APM_LINUX_POLLER_RATE);
```

Comment:
now I'm very confused. I'd expect the poller thread to sleep on the epoll_wait() call with the timeout. But here you are passing Hz and make the thread derived from SchedulerThread which effectively throttles the thread to run every 10msec (SchedulerThread is a PeriodicThread). Something seems wrong here.